community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStore.java

```diff
@@ -900,12 +900,32 @@ public long nextCommittingTransactionId()
     @Override
     public void transactionCommitted( long transactionId, long checksum )
     {
-        if ( lastCommittedTx.offer( transactionId, checksum ) )
+        checkInitialized( lastCommittingTxField.get() );
+        TransactionId current = highestCommittedTransaction.get();
+        if ( current.transactionId() < transactionId )
```

Comment:
if we will have 2 threads Tr1 with Tx = 2 and Tr2 with Tx = 3.
Then following sequence will cause an error:
```
+---------------------------------+-----------------------------------+
|              T1                 |                T2                 |
|                                 |                                   |
+---------------------------------------------------------------------+
|  passing if statement           |                                   |
|                                 |                                   |
|                                 |   passing if statement            |
|                                 |                                   |
|                                 |   do CAS                          |
|                                 |                                   |
|  do CAS                         |                                   |
|                                 |                                   |
+---------------------------------+-----------------------------------+
```